src/openms/source/FEATUREFINDER/MassTraceDetection.cpp

```
@@ -251,312 +254,570 @@
+              // bound indices are not matching (suggesting presence of multiple close neighbors
+              if (next_down_peak_idx_right != next_down_peak_idx_left && centroid_im != -1)
+              {
+                // generate a map of ion mobility value -- peak index pairs from mz peak neighbors
+                std::map<int, double> ionMobilityMap;
+                for (Size im_i = next_down_peak_idx_left; im_i <= next_down_peak_idx_right; ++im_i)
+                {
+                  ionMobilityMap[im_i] = spec_trace_down.getFloatDataArrays()[Ion_Mobility_idx][im_i];
+                }
+                // find the closest ion mobility value to centroid_im
+                // and update next_down_peak_idx
+                double closestValue = ionMobilityMap.begin()->second;
```

Cppcheck (2.10-2)

[error] Out of bounds access in expression 'ionMobilityMap.begin()' because 'ionMobilityMap' is empty.

(containerOutOfBounds)